src/ORM/DataObject.php

```diff
@@ -3896,28 +3933,40 @@ public function searchableFields()
         $rewrite = [];
         foreach ($fields as $name => $specOrName) {
             $identifier = (is_int($name)) ? $specOrName : $name;
+            $relObject = isset($specOrName['match_any']) ? null : $this->relObject($identifier);
```

Comment:
This was only being called if the spec is an array - but that means if you're trying to search against a field that isn't valid, you only get an exception if you're using complex config! Otherwise you have to filter to get the exception, which isn't very friendly.

Note that it _is_ possible for this to return `null`